packages/myst-to-jats/src/transforms/blocks.ts

```diff
@@ -1,28 +1,52 @@
 import type { Plugin } from 'unified';
 import type { Block } from 'myst-spec-ext';
-import type { GenericParent } from 'myst-common';
+import { selectBlockParts, type GenericParent } from 'myst-common';
 import { remove } from 'unist-util-remove';
 import { selectAll } from 'unist-util-select';
+import { ABSTRACT_PARTS, ACKNOWLEDGMENT_PARTS, type Options } from '../types.js';
+
+function blocksToKeep(tree: GenericParent, opts: Options) {
+  const keepParts: string[] = [...ACKNOWLEDGMENT_PARTS];
+  if (opts.extractAbstract) {
+    keepParts.push(...(opts.abstractParts?.map(({ part }) => part).flat() ?? ABSTRACT_PARTS));
+  }
+  if (opts.backSections) {
+    keepParts.push(...opts.backSections.map(({ part }) => part).flat());
+  }
+  return selectBlockParts(tree, keepParts);
+}
 
 /**
  * This transform does the following:
- * - Removes hidden or removed blocks from the tree
+ * - Removes hidden or removed blocks from the tree except those that are consumed as front/backmatter parts
  * - Removes hidden or removed outputs from the tree
  */
-export function blockTransform(tree: GenericParent) {
+export function blockTransform(tree: GenericParent, opts: Options) {
+  // Collect blocks that will be used as parts to make sure they are not deleted
+  const doNotDelete = blocksToKeep(tree, opts);
   // This could also be an output, etc.
   (selectAll('[visibility=remove],[visibility=hide]', tree) as Block[]).forEach((node) => {
     if (node.visibility === 'remove' || node.visibility === 'hide') {
       (node as any).type = '__delete__';
     }
   });
+  // Blocks are converted to sections - avoid doing this for part blocks
+  doNotDelete.forEach((node) => {
+    node.type = 'block-part' as any;
+  });
   const removed = remove(tree, '__delete__');
   if (removed === null) {
     // remove is unhappy if all children are removed - this forces it through
     tree.children = [];
   }
 }
 
-export const blockPlugin: Plugin<[], GenericParent, GenericParent> = () => (tree) => {
-  blockTransform(tree);
+export const blockPlugin: Plugin<[Options], GenericParent, GenericParent> = (opts) => (tree) => {
+  blockTransform(tree, opts);
 };
+
+export function restoreBlockPartTypeTransform(tree: GenericParent) {
+  selectAll('block-part', tree).forEach((node) => {
+    node.type = 'block';
```

Comment:
Finally, once transformations are done, we return them to their original `'block'` type, so the `extractParts` function will behave correctly. Pretty gross.

Reply:
I am leaving this grossness in, however, since I think the real fix is bulid JATS in unist (there's already some utility for this in [jats-xml](https://github.com/curvenote/jats/blob/main/packages/jats-utils/src/utils.ts)) - then, we will not need to do these extra jats-specific transforms on the _input_ MyST AST; we instead can apply simpler, more sensible transforms to the _output_ JATS AST.